### PR TITLE
TMA-928: delete temporary user profiles

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

File: spec/project/project_spec.rb [hunk 1]
```diff
@@ -38,24 +45,14 @@
...
   describe '#add_users' do
     it 'Adding user without domain should fail if it is not in the project' do
-      users = (1..5).to_a.map do
-        {
-          user: ProjectHelper.create_random_user(@client),
-          role: 'Admin'
-        }
-      end
-
+      users = ProjectHelper.ensure_users(client: @client, amount: 5, caller: $example_name).map { |u| { user: u, role: 'Admin' } }
```
Comment (yenkins) [inline spec/project/project_spec.rb]: Identical code found in project_spec.rb:48, project_spec.rb:54 (mass = 72)

File: spec/project/project_spec.rb [hunk 2]
```diff
@@ -38,24 +45,14 @@
...
     end
 
     it 'Adding users with domain should pass and users should be added to domain' do
-      users = (1..5).to_a.map do
-        {
-          user: ProjectHelper.create_random_user(@client),
-          role: 'Admin'
-        }
-      end
+      users = ProjectHelper.ensure_users(client: @client, amount: 5, caller: $example_name).map { |u| { user: u, role: 'Admin' } }
```
Comment (yenkins) [inline spec/project/project_spec.rb]: Identical code found in project_spec.rb:48, project_spec.rb:54 (mass = 72)

File: spec/helpers/project_helper.rb [hunk 3]
```diff
@@ -40,20 +40,30 @@ def delete_old_projects(opts = { :client => GoodData.connection })
           end
         end
 
-        def create_random_user(client, opts = {})
+        def ensure_users(opts = {})
+          caller = opts[:caller] ? CGI.escape(opts[:caller]) : rand(1e7)
```
Comment (yenkins) [inline spec/helpers/project_helper.rb]: Calls 'opts[:caller]' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline spec/helpers/project_helper.rb]: Refers to 'opts' more than self (maybe move it to another class?) (FeatureEnvy)

File: spec/helpers/project_helper.rb [hunk 4]
```diff
@@ -40,20 +40,30 @@ def delete_old_projects(opts = { :client => GoodData.connection })
...
           num = rand(1e7)
           login = opts[:login] || "gemtest#{num}@gooddata.com"
 
-          opts = {
+          data = {
             email: login,
             login: login,
             first_name: 'the',
-            last_name: num.to_s,
+            last_name: login.split('@').first,
```
Comment (yenkins) [inline spec/helpers/project_helper.rb]: Refers to 'login' more than self (maybe move it to another class?) (FeatureEnvy)

File: spec/helpers/project_helper.rb [hunk 5]
```diff
@@ -40,20 +40,30 @@ def delete_old_projects(opts = { :client => GoodData.connection })
...
             role: 'editor',
-            password: CryptoHelper.generate_password,
+            password: login.reverse,
             domain: ConnectionHelper::DEFAULT_DOMAIN
           }
-          GoodData::Membership.create(opts, client: client)
+          GoodData::Membership.create(data, client: opts[:client])
```
Comment (yenkins) [inline spec/helpers/project_helper.rb]: Refers to 'opts' more than self (maybe move it to another class?) (FeatureEnvy)

File: spec/helpers/project_helper.rb [hunk 6]
```diff
@@ -40,20 +40,30 @@ def delete_old_projects(opts = { :client => GoodData.connection })
           end
         end
 
-        def create_random_user(client, opts = {})
+        def ensure_users(opts = {})
```
Comment (yenkins) [inline spec/helpers/project_helper.rb]: Has approx 8 statements (TooManyStatements)

File: spec/helpers/project_helper.rb [hunk 7]
```diff
@@ -40,20 +40,30 @@ def delete_old_projects(opts = { :client => GoodData.connection })
...
+          caller = opts[:caller] ? CGI.escape(opts[:caller]) : rand(1e7)
+          amount = opts[:amount] || 1
+          usrs = amount.times.map do |i|
```
Comment (yenkins) [inline spec/helpers/project_helper.rb]: Has the variable name 'i' (UncommunicativeVariableName)

File: spec/spec_helper.rb [hunk 8]
```diff
@@ -73,8 +73,8 @@
       if self.class.metadata[:vcr]
         # replace parallel iterations with the serial one, since VCR can't handle parallel request matching correctly
         module Enumerable
-          def peach_with_index(*)
-            each_with_index
+          def peach_with_index(*, &y)
```
Comment (yenkins) [inline spec/spec_helper.rb]: Has the parameter name 'y' (UncommunicativeParameterName)

File: spec/project/project_spec.rb [hunk 9]
```diff
@@ -116,19 +115,19 @@
...
 
-      expect(@project.members.count).to eq 4
+      expect(@project.members.count).to eq additional_batch.count + 1
       expect(@project.member?(bill)).to be_truthy
       expect(@project.members?(users - additional_batch).any?).to be_falsey
     end
 
     it "Updates user's role in a project" do
-      users = (1..5).to_a.map { ProjectHelper.create_random_user(@client).to_hash }
+      users = ProjectHelper.ensure_users(client: @client, amount: 5, caller: $example_name).map &:to_hash
```
Comment (yenkins) [inline spec/project/project_spec.rb]: Lint/AmbiguousOperator: Ambiguous block operator. Parenthesize the method arguments if it's surely a block operator, or add a whitespace to the right of the `&` if it should be a binary AND.

File: spec/project/project_spec.rb [hunk 10]
```diff
@@ -166,11 +166,13 @@
 
   describe '#set_user_roles' do
     it 'Properly updates user roles as needed' do
-      users_to_import = @domain.users.drop(rand(100)).take(5).map { |u| { user: u, role: 'admin' } }
+      users =  ProjectHelper.ensure_users(client: @client, amount: 5, caller: $example_name)
```
Comment (yenkins) [inline spec/project/project_spec.rb]: Style/SpaceAroundOperators: Operator `=` should be surrounded by a single space.

Comment (yenkins) [inline spec/project/project_spec.rb]: Style/ExtraSpacing: Unnecessary spacing detected.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/187/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1023/ : FAILURE in 1m 26s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1292/ : SUCCESS in 2m 52s

File: spec/project/project_spec.rb [hunk 11]
```diff
@@ -38,24 +47,14 @@
...
   describe '#add_users' do
     it 'Adding user without domain should fail if it is not in the project' do
-      users = (1..5).to_a.map do
-        {
-          user: ProjectHelper.create_random_user(@client),
-          role: 'Admin'
-        }
-      end
-
+      users = ProjectHelper.ensure_users(client: @client, amount: 5, caller: $example_name).map { |u| { user: u, role: 'Admin' } }
```
Comment (yenkins) [inline spec/project/project_spec.rb]: Identical code found in project_spec.rb:50, project_spec.rb:56 (mass = 72)

File: spec/project/project_spec.rb [hunk 12]
```diff
@@ -38,24 +47,14 @@
...
     end
 
     it 'Adding users with domain should pass and users should be added to domain' do
-      users = (1..5).to_a.map do
-        {
-          user: ProjectHelper.create_random_user(@client),
-          role: 'Admin'
-        }
-      end
+      users = ProjectHelper.ensure_users(client: @client, amount: 5, caller: $example_name).map { |u| { user: u, role: 'Admin' } }
```
Comment (yenkins) [inline spec/project/project_spec.rb]: Identical code found in project_spec.rb:50, project_spec.rb:56 (mass = 72)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/189/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1025/ : FAILURE in 1m 32s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1294/ : SUCCESS in 1m 22s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/195/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1030/ : SUCCESS in 3m 06s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1299/ : SUCCESS in 4m 34s

**Comment by kubamahnert** [comment]
https://travis-ci.org/kubamahnert/gooddata-ruby/builds/387683465

**Pull request merged**